### PR TITLE
Move addon_update e2e test to framework

### PR DESCRIPTION
**Pull request by gmarek**
cc @ixdy

### CODE REVIEW

File: test/e2e/addon_update.go [hunk 1]
```diff
@@ -222,24 +215,15 @@ var _ = Describe("Addon update", func() {
 
 	AfterEach(func() {
 		// This test requires:
-		// - SSH
-		// - master access
+		// - SSH master access
 		// ... so the provider check should be identical to the intersection of
 		// providers that provide those capabilities.
-		if !providerIs("gce") {
```
Comment (ixdy) [inline test/e2e/addon_update.go]: why was this removed?

File: test/e2e/addon_update.go [hunk 2]
```diff
@@ -222,24 +215,15 @@ var _ = Describe("Addon update", func() {
 
 	AfterEach(func() {
 		// This test requires:
```
Comment (ixdy) [inline test/e2e/addon_update.go]: probably just remove this comment entirely

**Comment by ixdy** [comment]
LGTM besides nit

File: test/e2e/addon_update.go (hunk 2 above)
Comment (gmarek) [inline test/e2e/addon_update.go]: Done.

**Comment by ixdy** [comment]
@k8s-bot test this again, please
